components/common/NewTable.tsx

```diff
@@ -41,6 +43,10 @@ export default function NewTable({
     return (
       <>
         {data.map((row, i) => {
+          if (row === null) {
+            return;
+          }
```

Comment:
It would be cleaner to first filter data for null then this check is unnecessary. If it makes TS angry I'd prefer returning `null` here. Idk what React does with `undefined` as part of an array but it's more standard to use `null` I think.